### PR TITLE
fix(core): convert legacy-sanitized values to Trusted Types

### PR DESCRIPTION
**Pull request by bjarkler**
When an application uses a custom sanitizer or one of the
bypassSecurityTrust functions, Angular has no way of knowing whether
they are implemented in a secure way. (It doesn't even know if they're
introduced by the application or by a shady third-party dependency.)
Thus using Angular's main Trusted Types policy to bless values coming
from these two sources would undermine the security that Trusted Types
brings.

Instead, introduce a Trusted Types policy called
angular#unsafe-legacy-bypass specifically for blessing values from these
sources. (The term legacy is used as there will be a new way of
implementing sanitizers that support Trusted Types.) This allows an
application to enforce Trusted Types even when their sanitizer or
bypassSecurityTrust function calls have not been migrated, knowing that
compromises to either of these two sources may lead to arbitrary script
execution.

Applications that have migrated to the new Trusted Types compatible
sanitizers should not allow the angular#unsafe-legacy-bypass policy.

This is based on #39217. See the individual commits for more details.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [X] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Other information
This is part of an ongoing effort to add support for Trusted Types to Angular.

### CODE REVIEW

File: packages/core/src/render3/interfaces/renderer.ts [hunk 1]
```diff
@@ -157,9 +157,9 @@ export interface RElement extends RNode {
   classList: RDomTokenList;
   className: string;
   textContent: string|null;
-  setAttribute(name: string, value: string): void;
+  setAttribute(name: string, value: any): void;
```
Comment (IgorMinar) [inline packages/core/src/render3/interfaces/renderer.ts:160]: as discussed, please turn all of these new `any`s into `string|TrustedHTML|TrustedScript|TrustedScriptURL`. thanks!

**Review by IgorMinar — CHANGES_REQUESTED** [review]
the rest looks good to me.

File: packages/core/src/render3/interfaces/renderer.ts (hunk 1 above)
Comment (bjarkler) [reply packages/core/src/render3/interfaces/renderer.ts]: Fixed.

**Review by IgorMinar — APPROVED** [review]
this looks great to me now!

Reviewed-for: global-approvers, fw-security

**Comment by bjarkler** [comment]
There might still be typing issues with this PR (it modifies the ɵɵsanitize* instructions to return Trusted Types), so let's wait for the tests to pass before merging.

**Pull request closed without merging**